src/common/ina.rs

```diff
@@ -0,0 +1,246 @@
+/// Package ina implements ina power sensors to measure voltage, current, and power
+/// INA219 datasheet: https://www.ti.com/lit/ds/symlink/ina219.pdf
+/// INA226 datasheet: https://www.ti.com/lit/ds/symlink/ina226.pdf
+///
+/// The voltage, current and power can be read as
+/// 16 bit big endian integers from their given registers.
+/// This value is multiplied by the register LSB to get the reading in nanounits.
+///
+/// Voltage LSB: 1.25 mV for INA226, 4 mV for INA219
+/// Current LSB: maximum expected current of the system / (1 << 15)
+/// Power LSB: 25 * Current LSB for INA226, 20 * Current LSB for INA219
+///
+/// The calibration register is programmed to measure current and power properly.
+/// The calibration register is set to: calibratescale / (current_lsb * sense_resistor)
+use crate::common::i2c::I2CHandle;
+use core::fmt;
+use std::sync::{Arc, Mutex};
+
+use super::{
+    board::Board,
+    config::ConfigType,
+    i2c::I2cHandleType,
+    power_sensor::{Current, PowerSensor, PowerSensorType, PowerSupplyType, Voltage},
+    registry::{get_board_from_dependencies, ComponentRegistry, Dependency},
+    status::Status,
+};
+
+const DEFAULT_I2C_ADDRESS: u8 = 0x40;
+const DEFAULT_SHUNT_RESISTANCE_OHMS: f64 = 0.1;
+
+const INA_219_CALIBRATION_SCALE: f64 = 0.04096;
+const INA_226_CALIBRATION_SCALE: f64 = 0.00512;
+
+const DEFAULT_CONFIG_REGISTER_VALUE: u16 = 0x399F;
+const CONFIG_REGISTER: u8 = 0x00;
+const VOLTAGE_REGISTER: [u8; 1] = [0x02];
+const CURRENT_AMPERES_REGISTER: [u8; 1] = [0x04];
+const POWER_REGISTER: [u8; 1] = [0x03];
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
```

Comment:
An annoying question, I'll admit, but: given that we now have rudimentary module support for micro-rdk, do the INA power sensors belong as builtins, or should they be in a separate module?

Reply:
I think we should we keep as builtin drivers for now and move them out when we have an easier build process for modules?

Reply:
I don't think we have a clear view on whether or not we need these sensor as builtin, it would help having them since it make micro-rdk more versatile out of the box. As gautham said not having a process where people can generate an image just by adding modules makes it unlikely they will ever be used at this stage, plus I would like to have a stronger CI to protect against regression and module that start to fail to built against micro-rdk as versions get released. We are still changing interfaces and all after all.

Reply:
@npmenard - OK, sounds good. Resolving this comment.